framework/docker/docker_options.go

```diff
@@ -72,3 +72,94 @@ func WithPerLightNodeConfig(nodeConfigs map[int]*DANodeConfig) ConfigOption {
 		cfg.DataAvailabilityNetworkConfig.LightNodeConfigs = nodeConfigs
 	}
 }
+
+// SIMPLE: One-liner to configure all DA node ports
+func WithDANodePorts(rpcPort, p2pPort string) ConfigOption {
+	return func(cfg *Config) {
+		if cfg.DataAvailabilityNetworkConfig == nil {
+			cfg.DataAvailabilityNetworkConfig = &DataAvailabilityNetworkConfig{}
+		}
+		cfg.DataAvailabilityNetworkConfig.DefaultRPCPort = rpcPort
+		cfg.DataAvailabilityNetworkConfig.DefaultP2PPort = p2pPort
+	}
+}
+
+// SIMPLE: One-liner to configure core connection ports
+func WithDANodeCoreConnection(rpcPort, grpcPort string) ConfigOption {
+	return func(cfg *Config) {
+		if cfg.DataAvailabilityNetworkConfig == nil {
+			cfg.DataAvailabilityNetworkConfig = &DataAvailabilityNetworkConfig{}
+		}
+		cfg.DataAvailabilityNetworkConfig.DefaultCoreRPCPort = rpcPort
+		cfg.DataAvailabilityNetworkConfig.DefaultCoreGRPCPort = grpcPort
+	}
+}
+
+// SIMPLE: Configure everything in one call
+func WithCustomPortsSetup(daRPC, daP2P, coreRPC, coreGRPC string) ConfigOption {
```

Comment:
I'm not sure if this one is required, it's possible to do everything individually and it is more readable/obvious what is happening in the code if each is used individually I think. You would need to dig into this method to see the order also since it's just 4 strings.